### ISSUE TITLE
prom_push_url defined but http not available!

### ISSUE THREAD

**Issue by Andarius68** [minetest-monitoring/monitoring#33]
Oddly, the same setup is working on my test server with no issues but when I tried to do it on my main server I get the aforementioned error even though I have monitoring listed in minetest.conf among the http secure mods:

`#prometheus
monitoring.prometheus_push_url = https://monitoring.minetest.ch/push/metrics/job/minetest/instance/beul-aithris
monitoring.mesecons.verbose = false
prometheus_listener_address = 127.0.0.1:60000

#http security
secure.http_mods = monitoring,yl_matterbridge,xp_redo,digistuff`

Further details available here: https://forum.minetest.net/viewtopic.php?p=435563#p435563

At BlockHead's suggestion I tried installing the QoS mod on the main server to see if it would resolve the issue but it did not. This is a real head scratcher. Not sure what is different between my test server setup and the main server. But it's a no go.

**Comment by Andarius68** [comment]
Some updated info. Here's what I get in the log when I attempt to run it with QoS enabled:

`[MOD] Wine loaded
QoS control HTTP API request failed for monitoring
[monitoring] sampling: 100000 calls took 3727 us
[monitoring] registering builtin metrics
[monitoring] enabling digilines integrations
[monitoring] enabling mesecons integrations
[monitoring] enabling pipeworks integrations
[pipeworks_monitoring] intercepting can_go call for pipeworks:teleport_tube_1
[pipeworks_monitoring] intercepting can_go call for pipeworks:teleport_tube_2
[pipeworks_monitoring] intercepting can_go call for pipeworks:teleport_tube_3
[pipeworks_monitoring] intercepting can_go call for pipeworks:teleport_tube_7
[pipeworks_monitoring] intercepting can_go call for pipeworks:teleport_tube_10
[pipeworks_monitoring] intercepting can_go call for pipeworks:teleport_tube_9
[pipeworks_monitoring] intercepting can_go call for pipeworks:teleport_tube_8
[pipeworks_monitoring] intercepting can_go call for pipeworks:teleport_tube_6
[pipeworks_monitoring] intercepting can_go call for pipeworks:teleport_tube_5
[pipeworks_monitoring] intercepting can_go call for pipeworks:teleport_tube_4
<ACTION> 2024-04-29 12:32:58: [Main] Server: Shutting down
End log output over terminal (no stdout/stderr backlog during that)
========================
2024-04-29 12:32:59: ERROR[Main]: ModError: Failed to load and run script from /home/mtserver/5.9-dev-main/minetest/bin/../mods/monitoring/init.lua:
2024-04-29 12:32:59: ERROR[Main]: ...er/5.9-dev-main/minetest/bin/../mods/monitoring/init.lua:123: prom_push_url defined but http not available!
2024-04-29 12:32:59: ERROR[Main]: stack traceback:
2024-04-29 12:32:59: ERROR[Main]:       [C]: in function 'error'
2024-04-29 12:32:59: ERROR[Main]:       ...er/5.9-dev-main/minetest/bin/../mods/monitoring/init.lua:123: in main chunk
`

**Comment by BuckarooBanzay** [comment]
can you add the following snippet to any of your mods:

```lua
print("secure.http_mods = " .. minetest.settings:get("secure.http_mods"))
```

and observe if the output matches the configured setting.
(i'm 99% sure this is a configuration-issue :smirk:)

**Comment by Andarius68** [comment]
Hey, ya know what? Today I learned *again* that it's never a good idea to have multiple lines in your minetest.conf file that alter the same setting. When your snippet came back and said only yl_matterbridge had secure.http I thought "Well that's not right." So I went looking. Yep, there at the bottom of my minetest.conf file was another secure.http.mods line with only yl_matterbridge after it. So my other line that had not only yours but several other mods on it was completely ignored. Oops! Hehe. Yeah, make that 100% a configuration issue. Thank you for your time and consideration as well as for the many things  you do for the Minetest community. Including helping newbie idiots such as myself.
